File: src/Model/Student.ts
class Student extends User {

    private profRatings: ProfessorRating[];
    private UniRatings: UniversityRating[];
    private feedbacks: Feedback[];

    constructor(name: string, email: string, password: string) {
      super(name, email, password); // Chama o construtor da classe base (Usuario)
      this.profRatings = [];
      this.UniRatings = [];
      this.feedbacks = [];
    }
  
    public getProfRatings(): ProfessorRating[] {
      return this.profRatings;
    }

    public getUniRatings(): UniversityRating[] {
      return this.UniRatings;
    }

    public getFeedbacks(): Feedback[] {
      return this.feedbacks;
    }

    public setProfRatings(profRatings: ProfessorRating[]): void {
      this.profRatings = profRatings;
    }

    public setUniRatings(UniRatings: UniversityRating[]): void {
      this.UniRatings = UniRatings;
    }

    public setFeedbacks(feedbacks: Feedback[]): void {
      this.feedbacks = feedbacks;
    }

    public rateProf(): void {
      console.log(`${this.name} está avaliando um professor.`);

    }

    public rateUniversity(): void {
      console.log(`${this.name} está avaliando uma universidade.`);
      
    }
  
    public createFeedback(): void {
      console.log(`${this.name} está criando um feedback.`);
      
    }
}
  

File: src/main/java/com/unifei/ProfScore/entity/Student.java
package com.unifei.ProfScore.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Student extends User{

    @OneToMany(mappedBy = "student")
    private List<ProfessorRating> profRatings = new ArrayList<>();

    @OneToMany
    private List<UniversityRating> uniRatings = new ArrayList<>();

    @OneToMany
    private List<Feedback> feedbacks = new ArrayList<>();

    public Student(String name, String email, String password) {
        super(name, email, password);
    }

}
